Modules/System/AL Rest Client/src/ALHttpResponseMessage.Codeunit.al

```diff
@@ -0,0 +1,99 @@
+/// <summary>Holder object for the HTTP response data.</summary>
+codeunit 50302 "AL Http Response Message"
+{
+    var
+        HttpResponseMessage: HttpResponseMessage;
+        ALHttpContent: Codeunit "AL Http Content";
+        ErrorMessage: Text;
+        Initialized: Boolean;
+        HasResponse: Boolean;
+        NotInitializedErr: Label 'The Http Response Message has not been initialized';
+
+    internal procedure Initialize()
+    begin
+        Clear(HttpResponseMessage);
+        HasResponse := false;
+        Initialized := true;
+    end;
+
+    /// <summary>Indicates whether the HTTP response message has a success status code.</summary>
+    /// <returns>True if the HTTP response message has a success status code; otherwise, false.</returns>
+    /// <remarks>Any value in the HTTP status code range 2xx is considered to be successful.</remarks>
+    procedure IsSuccessStatusCode() Result: Boolean
+    begin
+        AssertInitialized();
+        if HasResponse then
+            Result := HttpResponseMessage.IsSuccessStatusCode;
+    end;
+
+    /// <summary>Gets the HTTP status code of the response message.</summary>
+    /// <returns>The HTTP status code.</returns>
+    procedure HttpStatusCode() ReturnValue: Integer
+    begin
+        AssertInitialized();
+        if HasResponse then
+            ReturnValue := HttpResponseMessage.HttpStatusCode
+    end;
+
+    /// <summary>Gets the reason phrase which typically is sent by servers together with the status code.</summary>
+    /// <returns>The reason phrase sent by the server.</returns>
+    procedure ReasonPhrase() ReturnValue: Text
+    begin
+        AssertInitialized();
+        if HasResponse then
+            ReturnValue := HttpResponseMessage.ReasonPhrase;
+    end;
+
+    /// <summary>Gets the HTTP content sent back by the server.</summary>
+    /// <returns>The content of the HTTP response message.</returns>
+    procedure Content() ReturnValue: Codeunit "AL Http Content"
+    begin
+        AssertInitialized();
+        if HasResponse then
+            ReturnValue := ALHttpContent;
+    end;
+
+    /// <summary>Sets the HTTP response message.</summary>
+    /// <param name="ResponseMessage">The HTTP response message.</param>
+    procedure SetResponseMessage(var ResponseMessage: HttpResponseMessage)
+    begin
+        AssertInitialized();
+        HttpResponseMessage := ResponseMessage;
+        ALHttpContent := ALHttpContent.Create(ResponseMessage.Content);
+        HasResponse := true;
```

Comment:
It's used as an internal flag to avoid reading values from an uninitialized HttpResponseMessage object.
There is a response if the status code has a non-zero value.
In theory, SetResponseMessage could be an internal procedure because it is normally only used after making the call. But for testing purposes if could be useful to have a public procedure.
And I don't think SetResponseMessage should return HasResponse. That does not have added value I'd say.